tests/quasi-newton/helpers.cpp

```diff
@@ -70,16 +77,22 @@ void runTestQN(std::string const &config, TestContext const &context)
 
```

Comment:
Could you please add here above some documentation on how primary and secondary data here is handled.